plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/preferences/PrefPageDatabaseEditors.java

```diff
@@ -165,25 +186,22 @@ public void widgetSelected(SelectionEvent e) {
             booleanNullPanel = new BooleanPanel(group, BooleanState.NULL);
         }
 
-        performDefaults();
+        notifyBooleanStylesChanged(BooleanStyleSet.getDefaultStyles(DBWorkbench.getPlatform().getPreferenceStore()));
 
         return composite;
     }
 
     @Override
-    protected void performDefaults()
-    {
-        DBPPreferenceStore store = DBWorkbench.getPlatform().getPreferenceStore();
-
-        keepEditorsOnRestart.setSelection(store.getBoolean(DBeaverPreferences.UI_KEEP_DATABASE_EDITORS));
-        keepEditorsOnDisconnect.setSelection(store.getBoolean(DBeaverPreferences.UI_KEEP_DATABASE_EDITORS_ON_DISCONNECT));
-        refreshEditorOnOpen.setSelection(store.getBoolean(NavigatorPreferences.NAVIGATOR_REFRESH_EDITORS_ON_OPEN));
-        editorFullName.setSelection(store.getBoolean(DBeaverPreferences.NAVIGATOR_EDITOR_FULL_NAME));
-        showTableGrid.setSelection(store.getBoolean(NavigatorPreferences.NAVIGATOR_EDITOR_SHOW_TABLE_GRID));
-        showPreviewOnSave.setSelection(store.getBoolean(NavigatorPreferences.NAVIGATOR_SHOW_SQL_PREVIEW));
-        syncEditorDataSourceWithNavigator.setSelection(store.getBoolean(NavigatorPreferences.NAVIGATOR_SYNC_EDITOR_DATASOURCE));
-
-        notifyBooleanStylesChanged(BooleanStyleSet.getDefaultStyles(store));
+    protected void performDefaults() {
+        keepEditorsOnRestart.setSelection(true);
+        keepEditorsOnDisconnect.setSelection(true);
+        refreshEditorOnOpen.setSelection(false);
+        editorFullName.setSelection(false);
+        showTableGrid.setSelection(true);
+        showPreviewOnSave.setSelection(true);
+        syncEditorDataSourceWithNavigator.setSelection(false);
```

Comment:
I really don't like the idea of having two places to store default values for preferences. I don't understand the point of doing that. Please, explain.

Reply:
Agreed. 
The idea is to use 
`store.getDefaultBoolean(DBeaverPreferences.UI_KEEP_DATABASE_EDITORS)` but not hardcoded constants in every page.